File: JiHwan/BJ/11723.py
import sys

n = int(input())

S = 0
for i in range(n):
    order = sys.stdin.readline().split()
    if order[0]=='empty':
        S = 0
    elif order[0]=='all':
        S =(1<<20)-1
    else:
        command = order[0]
        num = int(order[1])-1
        if command=='add':
            if S &(1<<num):
                pass
            S |= 1 << num

        elif command=='remove':
            if S & (1 << num):
                S &= ~(1 << num)
            else:
                pass
        elif command=='check': # x가 있으면 1 없으면 0
            if S & (1<<num):
                print(1)
            else:
                print(0)
        elif command=='toggle': #x가 있으면 x를 제거, 없으면 x를 추가
            S ^= 1<<num

File: SeongYeon/ts/baekjoon/11723.ts
// input
const input: string[] = require('fs')
  .readFileSync('test/input.txt')
  .toString()
  .trim()
  .split('\n');

// variable
let s: number = 0;

// solution
input.forEach((command, idx) => {
  if (idx !== 0) {
    const split: string[] = command.split(' ');
    const operator: string = split[0]!;
    const x: number = Number(split[1]);

    if (operator === 'add') {
      s |= 1 << (x - 1);
    } else if (operator === 'remove') {
      s &= ~(1 << (x - 1));
    } else if (operator === 'check') {
      if (s & (1 << (x - 1))) console.log(1);
      else console.log(0);
    } else if (operator === 'toggle') {
      s ^= 1 << (x - 1);
    } else if (operator === 'all') {
      s = (1 << 20) - 1;
    } else if (operator === 'empty') {
      s = 0;
    }
  }
});
